metrics-httpclient/src/main/java/com/codahale/metrics/httpclient/InstrumentedHttpClients.java

```diff
@@ -28,8 +28,7 @@ public static HttpClientBuilder custom(MetricRegistry metricRegistry,
                                            HttpClientMetricNameStrategy metricNameStrategy) {
         return HttpClientBuilder.create()
                 .setRequestExecutor(new InstrumentedHttpRequestExecutor(metricRegistry, metricNameStrategy))
-                .setConnectionManager(new InstrumentedHttpClientConnectionManager(metricRegistry));
+                .setConnectionManager(InstrumentedHttpClientConnectionManager.builder(metricRegistry).build());
```

Comment:
This change doesn't seem to be very useful, at the end we invoke the same constructor anyway...

Reply:
If we go for the builder pattern, we should discourage usage of the constructor. Ideally - make it private, which consequently makes this change mandatory. So even if we end up not making the constructor private - this would still be a good change as we kinda demo, how the library is to be used.

Performance-wise this is not a big deal, as setting up metrics usually happens only once per binary execution.